src/routes/user/_user.access-keys.tsx

```diff
@@ -0,0 +1,185 @@
+import { useSuspenseQuery } from '@tanstack/react-query';
+import { createFileRoute, useNavigate, useSearch } from '@tanstack/react-router';
+
+import { useEffect, useState } from 'react';
+
+import { z } from 'zod';
+
+import {
+  accessKeysColumns,
+  DataTable,
+  DateRangeFilter,
+  Card,
+  CardContent,
+  CardHeader,
+  CardTitle,
+  Toaster,
+  Tabs,
+  TabsContent,
+  TabsList,
+  TabsTrigger,
+  AddAccessKeyDialog,
+  RevokeKeyDialog,
+} from '@/components';
+
+import { useSpvWalletClient } from '@/contexts';
+import { addStatusField, getDeletedElements, getRevokedElements } from '@/utils';
+import { accessKeysQueryOptions } from '@/utils/accessKeysQueryOptions.tsx';
+
+export const Route = createFileRoute('/user/_user/access-keys')({
+  component: AccessKeys,
+  preSearchFilters: [
+    (search) => ({
+      ...search,
+    }),
+  ],
+  validateSearch: z
+    .object({
+      createdRange: z.object({ from: z.string(), to: z.string() }).optional().catch(undefined),
+      order_by_field: z.string().optional().catch('id'),
+      revokedRange: z.object({ from: z.string(), to: z.string() }).optional().catch(undefined),
+      sort_direction: z.string().optional().catch('desc'),
+      updatedRange: z.object({ from: z.string(), to: z.string() }).optional().catch(undefined),
+      page: z.number().optional().catch(1),
+      page_size: z.number().optional().catch(10),
+    })
+    .optional(),
+
+  loaderDeps: ({ search }) => {
+    return {
+      order_by_field: search?.order_by_field,
+      sort_direction: search?.sort_direction,
+      createdRange: search?.createdRange,
+      updatedRange: search?.updatedRange,
+      revokedRange: search?.revokedRange,
+      page: search?.page,
+      page_size: search?.page_size,
+    };
+  },
+  loader: async ({
+    context: { queryClient, spvWallet },
+    deps: { order_by_field, sort_direction, page, page_size, createdRange, revokedRange, updatedRange },
+  }) =>
+    await queryClient.ensureQueryData(
+      accessKeysQueryOptions({
+        spvWalletClient: spvWallet.spvWalletClient!,
+        order_by_field,
+        sort_direction,
+        createdRange,
+        updatedRange,
+        revokedRange,
+        page,
+        page_size,
+      }),
+    ),
+});
+
+export function AccessKeys() {
+  const [tab, setTab] = useState<string>('all');
+
+  const navigate = useNavigate({ from: Route.fullPath });
+
+  const { spvWalletClient } = useSpvWalletClient();
+
+  const { order_by_field, sort_direction, createdRange, updatedRange, revokedRange, page, page_size } =
+    useSearch({
+      from: '/user/_user/access-keys',
+    }) || {};
+
+  const { data: accessKeys } = useSuspenseQuery(
+    accessKeysQueryOptions({
+      spvWalletClient: spvWalletClient!,
+      order_by_field,
+      sort_direction,
+      createdRange,
+      updatedRange,
+      revokedRange,
+      page,
+      page_size,
+    }),
+  );
+
+  // const accessKeys = useLoaderData({ from: '/user/_user/access-keys' });
```

Comment:
Could we remove this commented-out line?

Reply:
Removed